src/libraries/Microsoft.PowerFx.Core/Utils/DPath.cs

```diff
@@ -107,16 +113,36 @@ private void AssertValid()
             Contracts.Assert(IsValid);
         }
 
-        internal DPath Parent => _node == null ? this : new DPath(_node.Parent);
-
-        internal DName Name => _node == null ? default : _node.Name;
-
-        internal int Length => _node == null ? 0 : _node.Length;
-
-        internal bool IsRoot => _node == null;
-
+        /// <summary>
```

Comment:
All of these should be internal, no need to expose them unless you've got a use for them. In general, access should be through ToDotted, operator[] or the Segments function.